### ISSUE TITLE
possible to provide how redisearch works like kvrocks so we can help contribute to the code?

### ISSUE THREAD

**Issue by hiqsociety** [RediSearch/RediSearch#2160]
possible to provide how redisearch works like kvrocks so we can help contribute to the code?
any write ups like so?
https://kvrocks.medium.com/design-complex-structure-on-rocksdb-993a00fd4f6d

**Comment by gkorland** [comment]
Did you check this doc? https://oss.redislabs.com/redisearch/Development/

**Comment by hiqsociety** [comment]
does clustering work with redisearch without any modifications?

**Comment by hiqsociety** [comment]
@gkorland the link you sent isnt very informative for the internal structures on how redisearch works

**Comment by gkorland** [comment]
> does clustering work with redisearch without any modifications?

Yes RediSearch does support cluster modes see https://github.com/RediSearch/RSCoordinator

**Comment by hiqsociety** [comment]
which command types are used with redisearch?
key / hash / z types?
possible to add more modules to redis while using redisearch?

**Comment by ashtul** [comment]
@hiqsociety 
RediSearch currently supports Hash and JSON (not GA) and we consider more types in the future.
You can load multiple modules onto a Redis server. RedisJSON and RediSearch work together this way.

**Comment by hiqsociety** [comment]
i would like to contribute code to improve but
do u have more information for the internal structures on how redisearch works written like so?
https://kvrocks.medium.com/design-complex-structure-on-rocksdb-993a00fd4f6d
